src/bijectors/named_bijector.jl

```diff
@@ -0,0 +1,231 @@
+abstract type AbstractNamedBijector <: AbstractBijector end
+
+forward(b::AbstractBijector, x) = (rv = b(x), logabsdetjac = logabsdetjac(b, x))
+
+#######################
+### `NamedBijector` ###
+#######################
+"""
+    NamedBijector <: AbstractNamedBijector
+
+Wraps a `NamedTuple` of key -> `Bijector` pairs, implementing evaluation, inversion, etc.
+
+# Examples
+```julia-repl
+julia> using Bijectors: NamedBijector, Scale, Exp
+
+julia> b = NamedBijector((a = Scale(2.0), b = Exp()));
+
+julia> x = (a = 1., b = 0., c = 42.);
+
+julia> b(x)
+(a = 2.0, b = 1.0, c = 42.0)
+
+julia> (a = 2 * x.a, b = exp(x.b), c = x.c)
+(a = 2.0, b = 1.0, c = 42.0)
+```
+"""
+struct NamedBijector{names, Bs<:NamedTuple{names}} <: AbstractNamedBijector
+    bs::Bs
+end
+
+@inline names_to_bijectors(b::NamedBijector) = b.bs
+
+@generated function (b::NamedBijector{names})(x::NamedTuple) where {names}
+    return :(merge(x, ($([:($n = b.bs.$n(x.$n)) for n in names]...), )))
+end
```

Comment:
Maybe just fully generate the resulting named tuple based on the type of `x` if you are defining it in a generated function anyway?

Reply:
Good suggestion :+1: Will do!

Reply:
Diddit!

Reply:
Second thought: might revert this change since `merge` is already implemented using `@generated`.

Reply:
Yeah let's not make the code more complicated than needed